java/solomiia_tymoshchuk/src/com/zoolatech/lecture4/tasks/_4/Unit.java

```diff
@@ -0,0 +1,93 @@
+package com.zoolatech.lecture4.tasks._4;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+public enum Unit {
+    GRAM("g", new Aliases("gram", "grams", "Gram", "gramms")),
+    KILOGRAM("kg", new Aliases("kilos", "kilo", "Kilograms", "kgs", "kilograms")),
+    OUNCE("oz", new Aliases("ounce", "ounces")),
+    POUND("lb", new Aliases("Pound", "lbs", "pound")),
+    TON("t", new Aliases("ton", "Ton", "tonnes", "tns"));
+
+
+    private final String abbreviation;
+    private final Unit.Aliases aliases;
+
+    Unit(String abbreviation, Aliases aliases) {
+        this.abbreviation = abbreviation;
+        this.aliases = aliases;
+    }
+
+    public String getAbbreviation() {
+        return abbreviation;
+    }
+
+    public static Unit getByAlias(String alias) {
+        for (Unit unit : Unit.values()) {
+            if (unit.hasAlias(alias.toLowerCase())) return unit;
```

Comment:
```
unit.name().equals(alias.toUpperCase())
```
OR
```
unit.name().equalsIgnoreCase(alias)
```
would allow you to get rid of the class `Aliases`